internal/adapter/provider/cliproxyapi_grok/adapter_test.go

```
@@ -132,6 +159,127 @@ func TestEnsureOpenAIStreamFinishBeforeDoneDoesNotDuplicateFinishReason(t *testi
+	}
+	grok := adapter.(*CLIProxyAPIGrokAdapter)
+	fakeExec := &fakeGrokExecutor{chunks: []executor.StreamChunk{
+		{Payload: []byte(`data: {"id":"chunk-1","object":"chat.completion.chunk","model":"grok-test","choices":[{"index":0,"delta":{"content":"hel"},"finish_reason":null}]}`)},
+		{Payload: []byte(`data: {"id":"chunk-2","object":"chat.completion.chunk","model":"grok-test","choices":[{"index":0,"delta":{"content":"lo"},"finish_reason":null}]}` + "\n\n")},
+		{Payload: []byte("data: [")},
+		{Payload: []byte("DONE]\n\n")},
+	}}
+	grok.executor = fakeExec
+
+	body := []byte(`{"model":"grok-test","stream":true,"messages":[{"role":"user","content":"say hello"}]}`)
+	req := httptest.NewRequest(http.MethodPost, "/v1/chat/completions", strings.NewReader(string(body)))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)